x/concentrated-liquidity/position.go

```diff
@@ -174,6 +177,14 @@ func (k Keeper) setPosition(ctx sdk.Context,
 	// Set the pool ID to position ID mapping.
 	key = types.KeyPoolPositionPositionId(poolId, positionId)
 	store.Set(key, sdk.Uint64ToBigEndian(positionId))
+
+	// Set the position ID to underlying lock ID mapping if underlyingLockId is provided.
+	key = types.KeyPositionIdForLock(positionId)
+	positionHasUnderlyingLock := k.positionHasUnderlyingLockInState(ctx, positionId)
+	if !positionHasUnderlyingLock && underlyingLockId != 0 {
+		// We did not find an underlying lock ID, but one was provided. Set it.
+		store.Set(key, sdk.Uint64ToBigEndian(underlyingLockId))
+	}
```

Comment:
Please cover this by tests. I think this one is important since there are a few indexes get updated here

Reply:
Added SetPosition test here https://github.com/osmosis-labs/osmosis/pull/4882/commits/77fad9302ff0238def4cec1b1dde1aa28800a994